### PR TITLE
Define 10 UDTs

### PR DESCRIPTION
**Pull request by xyzdelete**

### CODE REVIEW

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 1]
```diff
@@ -1,477 +1,473 @@
...
+
+
+//UDT 3) Cat
+struct Cat
+{
+    //5 properties:
+    // 1) name (std::string)
+    std::string name = "No name specified";
+    // 2) age (unsigned int)
+    int age;
```
Comment (matkatmusic) [inline main.cpp:247]: provide a default value for the age.  `0` is fine
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -1,477 +1,473 @@
...
+        std::string fuelType = "Liquid fuel";
+        std::string oxidiserType = "Liquid oxidiser";
+        unsigned int numOfPumps = 2;
+        unsigned int numOfCombustionChambers = 1;
+        unsigned int numOfExhaustNozzles = 1;
+
+        // 3 things it can do:
+        bool performSelfTests();
+        void produceThrust();
+        void burnFuel();
```
Comment (matkatmusic) [inline main.cpp:279]: What sort of parameters might these actions depend on? 
perhaps the amount of fuel..
the nozzle size..
Think of that when populating the function parameter lists
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -1,477 +1,473 @@
...
+    // 3) chipset (std::string)
+    std::string chipsetName = "CHD3411";
+    // 4) vendor (std::string)
+    std::string vendorName = "VendorABC";
+    // 5) model name (std::string)
+    std::string modelName = "Model324RQ";
+
+    // 3 things it can do
+    // 1) regulate voltage
+    void regulateVoltage();
```
Comment (matkatmusic) [inline main.cpp:320]: what's the input voltage range? 
these should be your function parameters
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -1,477 +1,473 @@
...
+    // 3) peak power (unsigned int)
+    unsigned int peakPowerLevel = 1500;
+    // 4) form factor (std::string)
+    std::string formFactor = "ATX";
+    // 5) operating temperature (int)
+    int operatingTemperature = 50;
+
+    // 3 things it can do:
+    // 1) convert the alternating current
+    bool convertCurrent(); // returns true if the current has been converted successfully and false otherwise
```
Comment (matkatmusic) [inline main.cpp:345]: what voltage is this current at?  how many amps? 
What are you converting it to? 
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (xyzdelete) [reply main.cpp]: Done.

File: main.cpp (hunk 2 above)
Comment (xyzdelete) [reply main.cpp]: Done.

File: main.cpp (hunk 3 above)
Comment (xyzdelete) [reply main.cpp]: Done.

File: main.cpp (hunk 4 above)
Comment (xyzdelete) [reply main.cpp]: Done.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**